pandas/tests/io/xml/test_to_xml.py

```diff
@@ -290,6 +293,42 @@ def test_index_false_rename_row_root(datapath, parser):
         assert output == expected
 
 
+def test_index_false_with_offset_input_index(parser):
+    """
+    Tests that the output does not contain the `<index>` field when the index of the
+    input Dataframe has an offset.
+
+    This is a regression test for issue #42458.
+    """
+
+    expected = """\
+<?xml version='1.0' encoding='utf-8'?>
+<data>
+  <row>
+    <shape>square</shape>
+    <degrees>360</degrees>
+    <sides>4.0</sides>
+  </row>
+  <row>
+    <shape>circle</shape>
+    <degrees>360</degrees>
+    <sides/>
+  </row>
+  <row>
+    <shape>triangle</shape>
+    <degrees>180</degrees>
+    <sides>3.0</sides>
+  </row>
+</data>"""
+
+    offset_geom_df = geom_df.copy()
+    offset_geom_df.index = RangeIndex(start=10, stop=13, step=1)
```

Comment:
Maybe parametrize test to add string index: 

```python
@pytest.mark.parametrize("offset_index", [list(range(10,13)), [str(i) for range(10,13)]])
def test_index_false_with_offset_input_index(parser, offset_index):
    ...
    offset_geom_df.index = offset_index
```

Reply:
> Do we need a new import for simple assignment: offset_geom_df.index = range(10, 13)?

The reason I didn't use `list(range(...))` directly is because then mypy complains about a wrong type assignment.